src/wallet/spend.cpp

```diff
@@ -105,6 +105,49 @@ void CoinsResult::clear()
     other.clear();
 }
 
+void CoinsResult::erase(std::set<COutPoint>& preset_coins)
+{
+    for (auto& it : coins) {
+        auto& vec = it.second;
+        auto i = std::find_if(vec.begin(), vec.end(), [&](const COutput &c) { return preset_coins.count(c.outpoint);});
+        if (i != vec.end()) {
+            vec.erase(i);
+            break;
+        }
+    }
+}
+
+void CoinsResult::shuffle(FastRandomContext& rng_fast)
+{
+    for (auto& it : coins) {
+        Shuffle(it.second.begin(), it.second.end(), rng_fast);
+    }
+}
+
+void CoinsResult::push_back(OutputType type, const COutput& out)
```

Comment:
In a0abea26692e63e9205ec68bdbc04cfad8c95a3d "refactor: add new helper methods"

nit: We prefer UpperCamelCase for function names.